MaterialDesignThemes.Wpf/PaletteHelper.cs

```diff
@@ -12,6 +13,30 @@ namespace MaterialDesignThemes.Wpf
 {
     public class PaletteHelper
     {
+        private static ResourceDictionary refinedBrushDictionary;
+        private static SwatchesProvider provider;
+
+		  public PaletteHelper()
+        {
+            provider = new SwatchesProvider();
+
+            // Null-propagation used to ensure that PalletteHelper can be mocked when Application.Current
+            // returns null within the context of a unit test.
+            refinedBrushDictionary = Application.Current?.Resources;
+        }
+
+        /// <summary>
+        /// Providing ability to support custom array of colour swatch in the defined assembly.
+        /// Increased performance on searching of colour brushes by limiting the resource dictionary search field.
+        /// </summary>
+        /// <param name="assembly"> The Assembly containing the custom colour swatches.</param>
+        /// <param name="brushDictionary">The Resource Dictionary containing the Material Design Brush definition.</param>
+        public PaletteHelper(Assembly assembly, ResourceDictionary brushDictionary = null)
```

Comment:
Given that the intent of this is to allow for more control over the `SwatchesProvider` that gets used. I would suggest having this constructor take in a `SwatchesProvider` rather than an `Assembly`. This will help reduce coupling between the two classes.

I would also suggest null checking the incoming value as well.